data_structures/heap/heap.py

```diff
@@ -53,7 +53,20 @@ def __repr__(self) -> str:
         return str(self.h)
 
     def parent_index(self, child_idx: int) -> int | None:
-        """return the parent index of given child"""
+        """
+        return the parent index of given child
+
+        >>> h = Heap()
+        >>> h.build_max_heap([103, 9, 1, 7, 11, 15, 25, 201, 209, 107, 5])
```

Comment:
Let's test with zero, negative integers, and floating point numbers.

Let's test `h.parent_index(12)` which is NOT in the heap.

We want to test how the algorithm works but also how it fails when given bad input.